cordformation/src/main/kotlin/net/corda/plugins/Dockerform.kt

```diff
@@ -162,17 +165,40 @@ open class Dockerform @Inject constructor(objects: ObjectFactory) : Baseform(obj
                         "ports" to listOf(dbPort)
                 )
 
+                // attach database dependency to the node service
                 service["image"] = dockerImage ?: "entdocker.software.r3.com/corda-enterprise-java1.8-${it.runtimeVersion().toLowerCase()}"
                 service["depends_on"] = listOf(dbHost)
+
+                // append persistence volume if it is required
+                if (persistent) {
+                    val dbDataVolume = dockerConfig.getString("dockerConfig.dbDataVolume")
+                    val volume = "$dbHost-volume"
+                    val volumeDir = "$nodeBuildDir/data"
+                    File(volumeDir).mkdirs()
```

Comment:
What if the directory cannot be created? Although IIRC the `mkdirs()` function is a bit tricky because it only returns `true` if it _actually creates_ the full directory path, and returns `false` if the directory path already exists. So to check that the directory is **really** there is more like:
```kotlin
volumeDir.mkdirs() || volumeDir.isDirectory()
```
You should probably throw an error if the directory doesn't exist and cannot be created either. Gradle provides both `InvalidUserDataException` and `InvalidUserCodeException` for situations like this.